File: laboratory work 4/solutions/code/2.cpp
#include <iostream>
#include <math.h>

using namespace std;

#define STEP 0.2
#define START -2
#define END 5



int main()
{
	string* con = new string("+");
	while (*con != "-")
	{
		if (*con == "+") {
			//first part
			cout << "-------------------------------------------";
			cout << endl << "|";
			cout.width(21);
			cout << "y|";
			cout.width(21);
			cout << "x|" << endl;
			cout << "-------------------------------------------" << endl;
			for (double i = START; i <= END; i += STEP) {
				cout << '|';
				cout.width(20);
				cout << tan(i / 3) * sin(i - 1.2) << "|";
				cout.width(20);
				cout << i << "|" << endl << "-------------------------------------------" << endl;
			}



			//second part
			cout << "-------------------------------------------";
			cout << endl << "|";
			cout.width(21);
			cout << "y|";
			cout.width(21);
			cout << "x|" << endl;
			cout << "-------------------------------------------" << endl;
			for (double i = START; i <= END; i += STEP) {
				cout << '|';
				cout.width(20);
				cout << sin(i) / log(i + 4) << "|";
				cout.width(20);
				cout << i << "|" << endl << "-------------------------------------------" << endl;
			}
		}
		cout << "type '+' to repeat or '-' to terminate "; cin >> *con;
	}
	delete con;

	return 0;
}

File: laboratory work 4/solutions/code/2.py
import math

STEP=0.2
START=-2
END=5

toContinueString='+';
while toContinueString!='-':
	if toContinueString=="+":
		print("---------------------------------------------------------");
		print("|"+" "*26+"y|"+" "*26+"x|");
		print("---------------------------------------------------------");
		i=START
		while i<=END:
			iString=str(i);
			number=str(math.tan(i / 3) * math.sin(i - 1.2));
			print("|"+" "*(27-len(number))+number+"|"+" "*(27-len(iString))+iString+"|");
			print("---------------------------------------------------------");
			i+=STEP;
		print();
		print();
		
		print("---------------------------------------------------------");
		print("|"+" "*26+"y|"+" "*26+"x|");
		print("---------------------------------------------------------");
		i=START
		while i<=END:
			iString=str(i);
			number=str(math.sin(i) / math.log(i + 4));
			print("|"+" "*(27-len(number))+number+"|"+" "*(27-len(iString))+iString+"|");
			print("---------------------------------------------------------");
			i+=STEP;
	toContinueString=input("type '+' to repeat or '-' to exit: ");
